### PR TITLE
add(mp)c advisor api monitoring procedure

### PR DESCRIPTION
**Pull request by Sims24**
## Description

New monitoring procedure to chech Container metrics through cAdvisor. 

## Target serie

- [ ] 20.04.x
- [x] 20.10.x
- [x] 21.04.x
- [x] 21.10.x (master)

### CODE REVIEW

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+Ce Pack comprend un Modèle d'Hôte et plusieurs Modèles de Services permettant d'observer le statut 
+d'un noeud et les performances de conteneurs s'y éxécutant. Les Services suivants sont disponibles:
+  * Container-Usage
+  * Container-Disk-IO
+  * Container-Traffic
+  * Node-Status
+
+### Règles de découverte
+
+Le Pack cAdvisor API propose différentes règles de découvertes de Service.
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:28]:
```suggestion
Le Pack cAdvisor API propose différentes règles de découverte de Service.
```

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+Pour superviser les métriques des conteneurs, il est nécessaire d'utiliser la fonctionnalité de 
+découverte de Service de Centreon. Voici un résumé des règles disponibles: 
+
+| Rule                                    | Description                                            |
+|-----------------------------------------|--------------------------------------------------------|
+| Cloud-cAdvisor-API-Container-Disk-IO    | Discover containers and monitor Disk-IO metrics        |
+| Cloud-cAdvisor-API-Container-Usage      | Discover containers and monitor CPU & RAM consumption  |
+| Cloud-cAdvisor-API-Container-Traffic    | Discover containers and monitor bandwidth utilization  |
+
+### Métriques colletées
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:39]:
```suggestion
### Métriques collectées
```

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  * Node-Status
+
+### Règles de découverte
+
+Le Pack cAdvisor API propose différentes règles de découvertes de Service.
+
+Pour superviser les métriques des conteneurs, il est nécessaire d'utiliser la fonctionnalité de 
+découverte de Service de Centreon. Voici un résumé des règles disponibles: 
+
+| Rule                                    | Description                                            |
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:33]: table is in English

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+|-----------------------------------------|--------------------------------------------------------|
+| Cloud-cAdvisor-API-Container-Disk-IO    | Discover containers and monitor Disk-IO metrics        |
+| Cloud-cAdvisor-API-Container-Usage      | Discover containers and monitor CPU & RAM consumption  |
+| Cloud-cAdvisor-API-Container-Traffic    | Discover containers and monitor bandwidth utilization  |
+
+### Métriques colletées
+
+<!--DOCUSAURUS_CODE_TABS-->
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:41]: the whole section is in English

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+By default, the --use-name flag will instance the metric with the <container_name>. 
+If you want to use the <container_id> instead, remove it from the EXTRAOPTIONS macro at the Service Template level.
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Prérequis
+
+### cAdvisor
+
+Un conteneur ou un démon cAdvisor doit être en cours d'execution et disponible. La
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:85]:
```suggestion
Un conteneur ou un démon cAdvisor doit être en cours d'exécution et disponible. La 
```

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+
+### cAdvisor
+
+Un conteneur ou un démon cAdvisor doit être en cours d'execution et disponible. La 
+documentation officielle permet de [déployer le nécessaire rapidement](https://github.com/google/cadvisor#quick-start-running-cadvisor-in-a-docker-container).
+
+### Flux réseaux
+
+Le Collecteur doit être en mesure de contacter le serveur hébergeant cAdvisor au travers 
+du port TCP/8080 port. Attention, selon la configuration le port peut être différent.
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:91]:
```suggestion
du port TCP/8080. Attention, selon la configuration le port peut être différent. 
```

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+### Flux réseaux
+
+Le Collecteur doit être en mesure de contacter le serveur hébergeant cAdvisor au travers 
+du port TCP/8080 port. Attention, selon la configuration le port peut être différent. 
+
+## Installation
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--Online IMP Licence & IT-100 Editions-->
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:97]: The name of the tabs is in English

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+
+1. Installer le Plugin sur tous les Collecteurs Centreon allant superviser des containers via cAdvisor:
+
+```bash
+yum install centreon-plugin-Cloud-cAdvisor-Api
+```
+
+2. Dans l'interface de Centreon, installer le Plugin Pack *cAdvisor API* depuis la page "Configuration > Plugin Packs > Manager"
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:105]:
```suggestion
2. Dans l'interface de Centreon, installer le Plugin Pack *cAdvisor API* depuis la page `Configuration > Plugin Packs > Manager`
```

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+
+<!--Offline IMP License-->
+
+1. Installer le Plugin sur tous les collecteurs Centreon devant superviser des containers via cAdvisor:
+
+```bash
+yum install centreon-plugin-Cloud-cAdvisor-Api
+```
+
+2. Installer le Plugin sur chaque Collecteur allant superviser des containers via cAdvisor:
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:115]: Points 1 and 2 are the same, apart from some lowercase/uppercase characters. Is that intended?

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+
+```bash
+yum install centreon-pack-cloud-cadvisor-api
+```
+
+3. Dans l'interface de Centreon, installer le Plugin Pack *cAdvisor API* depuis la page "Configuration > Plugin Packs > Manager"
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:121]:
```suggestion
3. Dans l'interface de Centreon, installer le Plugin Pack *cAdvisor API* depuis la page `Configuration > Plugin Packs > Manager`
```

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+### Host configuration
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:125]:
```suggestion
### Configuration de l'hôte
```

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+
+Ajouter un Hôte depuis le menu `Configuration > Hosts` et selectionner le Modèle `Cloud-cAdvisor-API`.
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:127]:
```suggestion
Ajouter un Hôte depuis le menu `Configuration > Hosts` et sélectionner le Modèle `Cloud-cAdvisor-API`.
```

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+
+Voici une description rapide des Macros de configuration disponibles : 
+
+| Mandatory | Macro                     | Description                                       | Default value                    |
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:131]: Table is in English

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+
+La commande devrait retourner un message de sortie similaire à:
+
+```bash
+OK: Container 'gray.eth0' Traffic In: 43.99 b/s, Traffic Out: 39.92 b/s | 'gray.eth0#container.traffic.in.bitspersecond'=43.99b/s;;;0; 'gray.eth0#container.traffic.out.bitspersecond'=39.92b/s;;;0;
+Container 'gray.eth0' Traffic In: 43.99 b/s, Traffic Out: 39.92 b/s
+```
+
+Cette commande contrôle le traffic entrant et sortant d'un conteneur (`--plugin=cloud::cadvisor::restapi::plugin --mode=traffic`).
+Le conteneur ciblé à le nom gray (`--filter-name='^gray$'`) et nous faisons en sorte d'utiliser son nom comme instance pour les
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:165]:
```suggestion
Le conteneur ciblé a le nom gray (`--filter-name='^gray$'`) et nous faisons en sorte d'utiliser son nom comme instance pour les 
```

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+    --plugin=cloud::cadvisor::restapi::plugin --mode=traffic \
+    --help
+```
+
+## Diagnostic des erreurs communes
+
+### UNKNOWN: curl perform error : Timeout was reached
+
+Quand cette erreur est affichée, cela signifie qu'il n'a pas été possible de rentrer en contact avec 
+l'API cAdvisor
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:181]:
```suggestion
l'API cAdvisor.
```

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+
+L'utilisation du flag `--debug` permet d'afficher des détails utiles au diagnostique.
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:183]:
```suggestion
L'utilisation du flag `--debug` permet d'afficher des détails utiles au diagnostic.
```

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md [hunk 17]
```diff
@@ -0,0 +1,190 @@
...
+
+Si vous utilisez un proxy, assurez-vous de l'avoir renseigné dans la Macro PROXYURL de l'Hôte. 
+
+### UNKNOWN: No containers found or no data available for this specific metric.
+
+Cette erreur signifie qu'aucune donnée n'est disponible pour la métrique demandée ou que le nom du conteneur 
+est erronné (typo ou conteneur arrêté).
```
Comment (cg-tw) [inline fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:190]:
```suggestion
est erronné (faute de frappe, ou conteneur arrêté).
```

**Review by cg-tw — CHANGES_REQUESTED** [review]

File: fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md (hunk 4 above)
Comment (Sims24) [reply fr/integrations/plugin-packs/procedures/cloud-cadvisor-api.md:41]: It's on purpose

**Review by cg-tw — APPROVED** [review]

**Pull request merged**